### DISCUSSION TITLE
why does Yaml multi line block dumper not accept \r\n ??

### DISCUSSION THREAD

**Discussion by Lobz** [symfony/symfony#46037] [Q&A]
Hi I'm trying to use Yaml to export/import some models in a user-friedly manner, and the users want to be able to use Markdown, so multi line blocks are a must. Buut the system I'm using uses \r\n as EOL by default, and for some reason the Yaml::dump function doesn't accept \r\n at all in multi-line blocks. Why is that?? There's a literal extra clause in the if to refuse using multiline if a \r\n is found anywhere:

```php
if (Yaml::DUMP_MULTI_LINE_LITERAL_BLOCK & $flags && \is_string($value->getValue()) && false !== strpos($value->getValue(), "\n") && false === strpos($value->getValue(), "\r\n")) {
```
Why is that? I was about to open a bug report when I saw this line of code and was like "wait this looks deliberate"

**Comment by stof** [comment]
This is because the YAML spec defines that line breaks, be them LF or CRLF, must be parsed as LF. So if your PHP string contains `\r\n` and we dump it as a line break, parsing it again would not produce the same string.
